### DISCUSSION TITLE
Server Streaming delivery confirmation

### DISCUSSION THREAD

**Discussion by wackazong** [hyperium/tonic#1212] [Q&A]
How can I make sure that a client has received a message that is sent to the client in a server streaming response? I want to make sure the message has actually reached the client or otherwise resend it in the stream. I know that I can implement Drop to react to the client disconnecting. I want a handler that is called when a message has been delivered successfully to the client.

**Comment by LucioFranco** [comment]
You would need to have the client send an ack message back. This isn't something that is supported out of the box with grpc.
